packages/client/src/classes/client.js

```diff
@@ -112,21 +122,25 @@ class Client {
 
     //Perform request
     const promise = new Promise((resolve, reject) => {
-      http(request, (error, response, body) => {
-
-        //Request error
-        if (error) {
+      axios(request)
+        .then(response => {
+          // Successful response
+          var parsedResponse = {
+            statusCode: response.status,
+            body: response.data,
+          };
+          return resolve([parsedResponse, response.data]);
```

Comment:
This is a potentially breaking change so I'd like some feedback if we think there should be a major version roll.

Previously, this returned `[RequestResponse, any]` and now it's returning `[{body: any, statusCode: number}, any]` which does not match the TS which is `[AxiosResponse, any]`. Ideally, this should not be exposing the underlying type because it's an implementation detail that may change in the future. Similar to the request type.

I think the types should be defined in this repo and not expose the Axios/Request lib types.

Reply:
I agree. I'll implement an interface similar to [what we have in twilio-node](https://github.com/twilio/twilio-node/tree/master/lib/http). I also agree that this should warrant a major version roll